### ISSUE TITLE
Generate frozen GPT-5.6 Sol Mathia-style intuitions for qwen-lean validation workloads

### ISSUE THREAD

**Issue by murillo128** [murillo128/mathia#57]
Research umbrella: #2
Conceptual-specialist line: #29
Intuition interface/harness: #31
Historical frontier-reference protocol: #32
Proof-aware oracle boundary (contrast only): #52
Mathia-v2 general design: #55
Downstream formal workload: `murillo128/qwen-lean#78`
Future planner work: `murillo128/qwen-lean#80` — **out of scope here**

# Goal

Generate and freeze one reusable **theorem-only frontier-reference intuition corpus** over the complete qwen-lean Dataset-v2 validation workloads that we will later use to measure the upper bound of the `intuition -> formal worker` channel.

This issue generates **Mathia-style natural-language intuitions only**. It does **not** run qwen-lean, DeepSeek, Lean, Mathia-v2, a planner, RL, or any downstream proof experiment.

The scientific reason is to create a clean strong-reference condition before evaluating Mathia itself:

```text
validation theorem
      |
      v
GPT-5.6 Sol / xhigh
      |
      v
frozen Mathia-style intuition
```

A later separate issue may compare:

```text
Q4 direct
GPT-5.6-Sol intuition -> Q4
Mathia intuition      -> Q4
```

so we can distinguish a weak intuition generator from a formal worker that cannot exploit good guidance.

This corpus is **evaluation/reference material**, not Mathia training data.

# Exact workload: all 650 validation theorems

Generate intuitions for the full validation population, with no subjective task filtering:

- `minif2f-valid-clean-v2`: **244 tasks**;
- `fresh-composition-valid-v2`: **406 tasks**;
- total: **650 tasks / target accepted intuitions**.

Do **not** inspect or include any final test workload.

Do not choose tasks based on qwen-lean success/failure, `c_i`, pass@64, mathematical domain, apparent difficulty, or any other outcome-dependent property. We want the complete validation population so later analyses can stratify by difficulty without post-hoc generator selection.

## Frozen qwen-lean lineage

The task population is already frozen independently of this issue.

Authoritative Dataset-v2 acceptance lineage:

```text
repo: murillo128/qwen-lean
Dataset-v2 accepted main commit:
67d0cdc13ce38c3847633fe9982d0c6e1473e8ed
```

Relevant source contract from `murillo128/qwen-lean#78`:

- `minif2f-valid-clean-v2` is the accepted clean 244-task validation set;
- `fresh-composition-valid-v2` is the deterministic view of canonical Dataset-v2 records with `provenance == synthetic` and `role == validation`, exactly 406 statements;
- validation oracle proofs are evaluator-only and must not be exposed here.

The completed Q4 n=64 evidence published by qwen-lean at:

```text
71222005f7d9093f31b321f2d1115f321dc75bb9
```

confirms the exact current validation memberships. Preserve and verify these ordered task-set identities when materializing the local snapshot:

```text
miniF2F ordered_task_ids_sha256:
cb5e5edba99dcb1fad61a1f5f88fd38e0061d122b54ea4e39a630a18138cce13

fresh-composition ordered_task_ids_sha256:
7d01d2878a996a0fc57df1eb634ba562cebe396002dc0bd1f617b3b2e7e80f0b
```

If the reconstructed task identities do not match these frozen hashes/counts, stop rather than generating against a different population.

# Critical blindness boundary

The intuition generator is theorem-only.

For each task it may receive only the public mathematical/formal problem presentation needed to understand the theorem, for example:

- stable task ID / workload identity as non-semantic metadata;
- exact Lean theorem declaration and its public import/preamble/context where required for meaning;
- the common intuition request below.

It must **not** receive:

- any source/oracle/reference proof;
- any qwen-lean candidate or Lean error;
- Q0/Q1/Q2/Q3/Q4 outcomes;
- pass@k / `c_i` / solved-vs-unsolved stratum;
- DeepSeek outcomes;
- Stage-9 capability-gap annotations;
- hints derived from known solutions;
- prior assistant/chat discussion of that theorem.

The generation process must be structurally unable to read these fields. Do not merely instruct the model to ignore them.

This differs deliberately from #52: #52 defines a **proof-aware oracle** condition. This issue is a **theorem-only frontier reference** and must never see the proof.

# Generator contract

Reuse the strong-reference discipline already established in #32 as closely as possible.

Primary generator:

```text
model: gpt-5.6-sol
reasoning effort: xhigh
one fresh isolated ephemeral session per theorem
no tools
no browsing/web
no repository/project instructions injected into the model session
no user configuration/history
no conversation reuse across theorems
```

Record the exact Codex CLI/runtime/model configuration actually used. If an exact historical CLI version from #32 is unavailable, do not silently substitute semantics: record the actual version while preserving the frozen model/reasoning/session contract above.

## Exact intuition request

Use the accepted compact strategic request from the Mathia intuition-fertility line:

> Propose one compact mathematical strategy for why the result should hold and how a proof might be organized. Identify the main mechanism or representation and a small number of useful intermediate mathematical goals if needed. Mention an obstruction or essential assumption only if it materially guides the route. Do not write the proof.

The desired output is a **natural-language mathematical intuition**, not a Lean plan and not a proof.

Allowed:

- mathematical concepts and mechanisms;
- equations/ordinary mathematical notation;
- representation changes;
- useful intermediate mathematical goals described conceptually;
- symmetry/invariance/decomposition/contradiction/induction ideas at a mathematical level;
- essential assumptions or obstructions when genuinely useful.

Disallowed:

- Lean code or code fences;
- tactic scripts;
- proof terms;
- `by` blocks or executable fragments;
- mathlib API/lemma-name recipes;
- step-by-step near-complete formal proof transmission;
- copying formal identifiers as implementation instructions.

# Length / interoperability contract

Keep the generated intuition compatible with the existing Mathia/qwen-lean intervention interface from #31/#32.

Use a hard maximum of **96 tokens under the frozen qwen-lean tokenizer identity used by the accepted intuition-fertility interface**, measured on the exact downstream-visible intuition text before wrapper delimiters.

Do not semantically truncate an over-budget output. Reject it under the leakage/eligibility policy below.

Preserve exact raw text before any classification plus exact downstream-visible accepted text/hash. Do not rewrite/summarize/sanitize a generated intuition into compliance.

# Leakage / eligibility gate

Because the generator never sees a source proof, this issue needs the generic Mathia formal-leakage boundary, not #52's source-proof-overlap check.

Use mechanically auditable reason codes at minimum:

- `accepted`;
- `rejected_lean_syntax`;
- `rejected_formal_identifier`;
- `rejected_proof_like`;
- `rejected_over_budget`;
- `generation_failure`.

Do not use an LLM quality judge as the primary gate. The gate is about interface/proof leakage and budget compliance, **not whether the intuition is correct or insightful**.

Rejected outputs are preserved as raw evidence and are never repaired.

## Bounded regeneration policy

To maximize corpus completeness without outcome-based cherry-picking:

- attempt 1 is always generated once;
- if and only if attempt 1 fails the deterministic eligibility gate, allow **one** fresh isolated regeneration for that same theorem under the identical prompt/config;
- accept the first eligible attempt;
- maximum 2 attempts per theorem;
- if both fail, retain the theorem as `missing_accepted_intuition` with both raw attempts/reasons;
- no retry may depend on qwen-lean/Lean/DeepSeek outcomes or subjective intuition quality.

Freeze this policy before the first generation.

# Materialization / artifacts

Keep this experiment self-contained and inspectable under a bounded directory such as:

```text
experiments/frontier_intuition_corpus_v1/
  README.md
  source_tasks.jsonl
  generation_manifest.json
  prompts.jsonl
  raw_attempts.jsonl
  accepted_intuitions.jsonl
  leakage_results.jsonl
  summary.json
  freeze.json
```

Exact names may vary if an existing simple Mathia convention is better; do not create a new general dataset platform/DSL.

## `source_tasks` must contain

For every one of the 650 tasks, enough public information to reproduce generator input plus:

- workload (`minif2f-valid-clean-v2` or `fresh-composition-valid-v2`);
- exact task ID;
- declaration/public context;
- upstream qwen-lean commit/hash provenance;
- stable hash over exact model-visible theorem material.

It must contain **no proof and no qwen-lean/DeepSeek outcome metadata**.

## Every generation attempt must retain

- workload/task ID;
- attempt index;
- exact generator model/runtime identity;
- exact prompt text/hash;
- raw output text/hash;
- post-render/token-count evidence;
- leakage/eligibility result and reason;
- accepted-attempt identity where applicable.

The accepted-intuition artifact must be a deterministic projection from the raw attempts + frozen gate, not a hand-edited file.

# Evaluation-only / anti-contamination invariant

Every task and intuition in this corpus must be marked **`evaluation_only` / reference-only**.

These 650 frontier-reference outputs must **not** enter:

- Qwen-Mathia-v2 training (#55 or its execution issue);
- later Mathia SFT/RL corpora;
- qwen-lean training;
- planner training targets as teacher/gold plans.

They may later be used as a frozen **reference/upper-bound intervention condition** in a separately preregistered experiment.

This boundary is important: training Mathia on these exact intuitions would contaminate the future Base-vs-Mathia-vs-frontier comparison.

# Validation / audit

Before generation:

1. materialize exactly 244 + 406 = 650 source tasks;
2. verify no test-role task is present;
3. verify no proof/oracle field is present;
4. verify ordered task-ID hashes match the frozen qwen-lean identities above;
5. freeze exact prompt/config/regeneration policy.

After generation:

- report accepted/rejected/missing counts by workload and reason;
- report token-length distribution;
- report duplicate exact/normalized intuition rate as a diagnostic;
- prove every accepted row derives from a stored raw attempt;
- prove no raw attempt was overwritten;
- prove no qwen-lean/DeepSeek outcome metadata entered prompts or generator-visible artifacts;
- run a fresh independent read-only contract audit on the exact final artifact/freeze.

Do **not** select/regenerate intuitions because they sound weak or strong. Semantic quality is intentionally left for the later downstream experiment.

# Execution sequencing

This issue is intentionally runnable **while `murillo128/qwen-lean#78` finishes**.

It needs only the already-frozen validation task population, not the unfinished final Base/Q4/DeepSeek assessment or Stage-9 conclusions.

Do not wait for #78 to close. Conversely, do not read newly appearing #78 per-task outcomes during generation.

# Out of scope

- qwen-lean/Q4 inference;
- DeepSeek inference;
- Lean verification;
- Mathia-v1/v2 inference;
- planner inference/training;
- RL/GRPO;
- proof-aware oracle intuitions from #52;
- source-proof use;
- final-test tasks;
- filtering tasks by Q4 difficulty;
- semantic ranking or hand-picking generated intuitions;
- using these records as Mathia training data.

# Deliverable / exit gate

Close only when one immutable `frontier-intuition-corpus-v1` release contains:

- exact frozen 650-task source snapshot with qwen-lean lineage/hashes;
- frozen GPT-5.6 Sol / xhigh generation contract;
- one accepted intuition per task where the predeclared maximum-two-attempt policy succeeds;
- all raw attempts and eligibility decisions preserved;
- no source proof or downstream outcome leakage;
- no final-test membership;
- explicit `evaluation_only` boundary;
- deterministic hashes/freeze sufficient for a later qwen-lean/Mathia A/B issue;
- fresh independent audit with no material contract defect.

Record one final decision:

- `FRONTIER_INTUITION_CORPUS_READY`;
- `FRONTIER_INTUITION_GENERATION_BLOCKER`;
- `FRONTIER_INTUITION_INTEGRITY_BLOCKER`.

No downstream proof claim is authorized by completing this issue.

**Comment by murillo128** [comment]
## Hard isolation invariant — frontier intuition corpus is not part of Mathia training data

This issue must keep `frontier-intuition-corpus-v1` **physically and logically isolated** from every existing or future Mathia training corpus/release.

### Storage / lineage separation

- Store the #57 artifacts under a dedicated experiment/evaluation-only path (for example `experiments/frontier_intuition_corpus_v1/`), **not** under the canonical agnostic Mathia corpus, OpenAlex supplement, Riemann corpus, or any trainable corpus release directory.
- Do not append, merge, union, deduplicate-into, or otherwise materialize these records inside:
  - `agnostic-mathia-full-v1` / #44 lineage;
  - `agnostic-mathia-openalex-supplement-v1`;
  - the Riemann-Mathia corpus lineage;
  - the frozen `G-v2` optimizer manifest designed by #55;
  - any `mathia-interchange-v1` trainable manifest or successor training manifest.
- The qwen-lean synthetic validation declarations are copied/referenced only as **evaluation inputs/provenance** for generating frontier intuitions. They do not become Mathia source/interpretation/synthesis training objects.

### Mandatory metadata

Every #57 task/intuition record must carry an explicit non-trainable marker such as:

```text
evaluation_only: true
training_eligible: false
artifact_role: frontier_reference
```

Use the repository's existing field conventions if equivalent fields already exist; do not create duplicate semantics merely for naming consistency.

### Anti-contamination checks

Before freezing the release, mechanically verify and record that:

1. none of the 650 task IDs / generated intuition IDs are members of any canonical Mathia trainable manifest;
2. no #57 output path is consumed by the #55 G-v2 corpus materializer/training-manifest builder;
3. the frozen G-v2 optimizer manifest remains byte/hash-identical whether the #57 directory exists or is absent;
4. no source/interpretation/synthesis role in existing Mathia releases is assigned to #57 records;
5. any future code that enumerates trainable Mathia records ignores `evaluation_only` / `training_eligible=false` records by construction;
6. the final independent audit treats any leakage from #57 into Mathia training data as a **material integrity failure**, not a warning.

### Scientific reason

The future comparison needs the frontier intuition corpus to remain an external upper-bound/reference condition. If Mathia-v2 or a later Mathia checkpoint trains on these exact 650 GPT-5.6 Sol intuitions, the intended `Mathia theorem-only vs frontier theorem-only` comparison becomes contaminated and uninterpretable.

This isolation requirement is blocking for `FRONTIER_INTUITION_CORPUS_READY`.

**Comment by murillo128** [comment]
## Execution result — `FRONTIER_INTUITION_INTEGRITY_BLOCKER`

Issue #57 has been executed through generation, immutable capture/finalization, publication, and fresh independent review.

Evidence PR: #58  
Exact ready-for-review head: `267ed0a393c279de11086f40814a58e5b33256fa`

Frozen run:

- exact 650-task panel: 244 miniF2F + 406 fresh composition;
- 1,289 attempts under the maximum-two policy;
- 22 outputs accepted by the frozen gate (22 miniF2F, 0 fresh);
- 628 missing;
- generation freeze `frontier_intuition_corpus_4ec023f14ea83cac12b5ff9b742ecbbdaeb11be1f002fd7c5e7d1e69f0f40804`.

The required first independent audit returned **FAIL** and superseded the generation-only blocker decision:

1. 203 attempts contained unexpected completed tool items (712 `web_search`, one `todo_list`); 102 were terminal attempt-2 failures, so the runtime was not structurally tool-free.
2. The deterministic gate admitted forbidden Lean-plan/formal-instruction and near-complete-proof text among the 22 frozen-gate acceptances.

The evidence was preserved byte-for-byte; no output was relabeled post hoc. `independent_review.json` records the audited final decision `FRONTIER_INTUITION_INTEGRITY_BLOCKER`, forbids downstream use, and requires a newly frozen genuinely tool-unavailable runtime plus a boundary-complete gate before rerun.

A second fresh review of the exact blocker-recording head returned **PASS** for user review as blocker evidence. This is not a corpus-readiness verdict and not merge authorization.

Validation passed: exact source/freeze replay, 17 issue tests, 143 repository tests, and the independent reviewer's full 271-test discovery run.

Per the issue exit rule, #57 remains open and no downstream proof/fertility run is authorized.
